File: app/src/main/java/com/cv/streamingserver/WatchActivity.java
package com.cv.streamingserver;
import android.net.Uri;
import android.os.Bundle;
import android.widget.MediaController;
import android.widget.VideoView;
import androidx.appcompat.app.AppCompatActivity;

public class WatchActivity extends AppCompatActivity {

    private VideoView videoView;
    private MediaController mediaController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_watch);

        videoView = findViewById(R.id.videoView);
        mediaController = new MediaController(this);

        // Set media controller to the video view
        videoView.setMediaController(mediaController);

        // Set video URL from web address
        String videoUrl = getIntent().getStringExtra("vid");//"http://192.168.1.35/watch/6/";
        Uri uri = Uri.parse(videoUrl);
        videoView.setVideoURI(uri);

        // Start video playback
        videoView.start();
    }
}

File: app/src/main/java/com/cv/streamingserver/WatchActivity.kt
package com.cv.streamingserver;
import android.net.Uri
import android.os.Bundle
import android.widget.MediaController
import android.widget.VideoView
import androidx.appcompat.app.AppCompatActivity

class VideoActivity : AppCompatActivity() {

        override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_watch)

        val videoView = findViewById<VideoView>(R.id.videoView)

        // URL of the video to play
        val videoUrl = "http://192.168.1.35/watch/6/"

        // Create a MediaController
        val mediaController = MediaController(this)
        mediaController.setAnchorView(videoView)

        // Set the MediaController for the VideoView
        videoView.setMediaController(mediaController)

        // Set the URI of the video to be played
        videoView.setVideoURI(Uri.parse(videoUrl))

        // Start playing the video
        videoView.start()
        }
        }
